docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc

```diff
@@ -173,6 +173,147 @@ It is also possible to configure the scope values that would be used via the sco
+allows you to specify if you want to enable this using the `authentication-request-format` attribute.
+
+The `authentication-request-format` attribute has values of type `String` and has 3 allowed values:
+
+* `oauth2`: This sends the authentication request as usual following the oauth2 syntax. Parameters, such as,
+`state` are added to the authentication request directly.
+* `request`: This sends the authentication request as a signed and optionally encrypted JWT, which the OpenID provider
+can verify and/or decrypt. The OpenID provider must support request parameters, which is indicated using the
+`request_parameter_supported` field inside the OpenID metadata.
+* `request_uri`: This also sends the authentication request as a signed and optionally encrypted JWT, but `request_uri`
+sends the JWT by reference, as opposed to by value. The OpenID provider must support the `request_uri` parameter, indicated
+using the `request_uri_parameter_supported` field inside the OpenID metadata.
+
+NOTE: The default value for `authentication-request-format` is `oauth2`. If the OpenID provider does not support request
+parameters, then the request will be sent following the OAuth2 format and a warning will appear on the log messages.
+
+=== Passing Request Object by Value
+
+It is possible to send request objects by value when securing apps with OIDC. To do this, `authentication-request-format`
+must be set to `request`. The values for `request_type`, `client_id` and `redirect_uri` must be included in the request
+outside the JWT. All other parameters are added to the JWT and are passed by value.
+
+==== Signing Configuration
+The JWT containing the Request Object can be signed using one of the algorithms supported by the OpenID Provider, or can be sent as
+plaintext, indicated by the `none` algorithm in the JWS Header. The signature algorithm for the request object can be
+specified using the `request-object-signing-algorithm` attribute. This attribute is of type string and the default value
+is `none`, indicating that the request object will be sent as plaintext. However, this is just a default configuration
+and should not be used in production.
+
+If any asymmetric signing algorithm is specified as the signing algorithm for the request object, then keystore attributes
+should be configured. This is done using the following attributes:
+
+* `request-object-signing-keystore-file`: The absolute path to the client keystore file.
+* `request-object-signing-keystore-password`: The password to access the keystore file.
+* `request-object-signing-key-password`: The password to access the key within the keystore file.
+* `request-object-signing-key-alias`: The alias by which the key identified with within the keystore file.
+* `request-object-signing-keystore-type`: The keystore type for the file. This can be either "PKCS12" or "JKS".
+
+For `none` algorithm, we are not signing the JWT, so we don't need to configure a keystore. Similarly, for any HMAC
```

Comment:
When <attribute name> is set to `none`, the JWT won't be signed so there's no need to configure a keystore in this case.